### PR TITLE
OCPBUGS-7989: machine's node must be ready for CPMS machine to be ready

### PR DESCRIPTION
**Pull request by damdo**
A CPMS Machine should be considered Ready only when both its underlying MAPI Machine is `Running` **and** the linked Node is `Ready`.

### CODE REVIEW

File: test/integration/machine_manager.go [hunk 1]
```diff
@@ -201,6 +212,47 @@ func (r *integrationMachineManager) setPhase(ctx context.Context, logger logr.Lo
...
+					// The node must be Ready for the CPMS replica
+					// to be considered Ready.
+					Type:   corev1.NodeReady,
+					Status: corev1.ConditionTrue,
+				},
+			},
+		},
+	}
+
+	if err := r.Create(ctx, &node); err != nil {
```
Comment (JoelSpeed) [inline test/integration/machine_manager.go:239]: Does a node not have a status subresource? Do you not need to first create the node and then update the status on it?

Comment (damdo) [reply test/integration/machine_manager.go:239]: Yes, the Node has a status subresource, although I noticed that creating a Node with the status already filled in, is also a viable option, as conditions are retained.

**Comment by damdo** [comment]
Azure is expected to fail due to an existing issue independent of this PR: [OCPBUGS-7359](https://issues.redhat.com/browse/OCPBUGS-7359)

**Pull request merged**